cli/internal/plugins/git/versioner.go

```diff
@@ -0,0 +1,70 @@
+package git
+
+import (
+	"bytes"
+	"context"
+	"os/exec"
+	"strings"
+
+	"github.com/gobuffalo/buffalo-cli/plugins"
+)
+
+type Versioner struct {
+	pluginsFn plugins.PluginFeeder
+}
+
+func (b *Versioner) WithPlugins(f plugins.PluginFeeder) {
+	b.pluginsFn = f
+}
+
+func (b *Versioner) ScopedPlugins() []plugins.Plugin {
+	if b.pluginsFn == nil {
+		return []plugins.Plugin{}
+	}
+
+	var scoped []plugins.Plugin
+	for _, p := range b.pluginsFn() {
+		switch p.(type) {
+		case CommandRunner:
+			scoped = append(scoped, p)
+		}
+	}
+
+	return scoped
+}
+
+func (b *Versioner) BuildVersion(ctx context.Context, root string) (string, error) {
+	if _, err := exec.LookPath("git"); err != nil {
```

Comment:
@markbates I feel this should be part of the `cmdRunnerFn`. That way we could achieve independence from git when using the test `commandRunner`.

Reply:
I removed the check, as its unneeded. when you try to run the command it'll return an error if it can't find it.

Reply:
Will do the same in the Bzr tests.